janus_core/repository/invenio.py

```diff
@@ -0,0 +1,635 @@
+        requests.Request
+            Status of operation.
+        """
+        file = Path(file)
+
+        with file.open("rb") as in_file:
+            return _check(
+                requests.put(
+                    f"{self.bucket_url}/{self.name}",
+                    params={**params, "access_token": self.api_key},
+                    data=in_file,
+                ),
+                f"Uploading file {self.name} to deposition {self.dep_id}"
+            )
+
+class _Files(_SubCommandHandler):
+    """Handler for files within a deposition."""
+
+    def __init__(self, parent):
+        super.__init__(self, parent)
+
+    @property
+    def dep_id(self) -> str:
+        """Get deposition ID.
+
+        Returns
+        -------
+        str
+            Deposition ID.
+        """
+        return self.parent.dep_id
+
+    @property
+    def bucket_url(self) -> str:
+        """Get URL for new API file bucket.
+
+        Returns
+        -------
+        str
+            File bucket to ``put`` files.
+        """
+        return self.parent.bucket_url
+
+    def __getitem__(self, name) -> _File:
+        return _File(self, name)
+
+    def list(self, **params) -> requests.Request:
+        """Get information about all files in deposition.
+
+        Parameters
+        ----------
+        **params
+            Extra params for requests.
+
+        Returns
+        -------
+        requests.Request
+            Information about operation state.
+        """
+        return _check(
+            requests.get(
+                f"{self.url}/deposit/depositions/{self.dep_id}/files",
+                params={**params, "access_token": self.api_key},
+            ),
+            f"listing deposition {self.dep_id} files",
+        )
+
+    def sort(self, sorted_ids: dict[str, str], **params) -> requests.Request:
+        """Re-order files in deposition.
+
+        Parameters
+        ----------
+        sorted_ids
+            IDs of re-sorted files.
+
+        Returns
+        -------
+        requests.Request
+            Status of operation.
+        """
+        return _check(
+            requests.put(
+                f"{self.url}/deposit/depositions/{self.dep_id}/files",
+                params={**params, "access_token": self.api_key},
+                data=json.dumps(sorted_ids),
+                headers={"Content-Type": "application/json"},
+            ),
+            f"sorting files for deposition {self.dep_id}"
+        )
+
+    def upload(self, files: dict[str, Path], **params) -> requests.Request:
+        """Upload a set of files to a deposition.
+
+        Parameters
+        ----------
+        files
+            Dictionary where the key is the name for the repo,
+            and the value is a path to the file to upload.
+
+        Returns
+        -------
+        requests.Request
+            Status of operation.
+        """
+        request_list = []
+        for name, file in files.items():
+            file = Path(file)
```

Comment:
What if the path is a directory?

(Do we want to be able upload all files in a directory?)